Krypt/Source/pkcs8.c

```diff
@@ -32,3 +37,57 @@ char *convert_pkcs1_to_pkcs8(const char *pem) {
     
     return str_from_BIO(out);
 }
+
+char *pkcs8_encrypt(const char *pkcs1, const char *password) {
+  BIO *out = NULL;
+  EVP_PKEY *pkey = NULL;
+  PKCS8_PRIV_KEY_INFO *p8inf = NULL;
+  X509_ALGOR *pbe = NULL;
+  X509_SIG *p8 = NULL;
+
+  const EVP_CIPHER *cipher = EVP_aes_256_cbc();
+  int iter = 100000;
+  int pbe_nid = NID_hmacWithSHA256;
+
+  unsigned long int passlenuint = strlen(password);
+  int passlen = passlenuint & INT_MAX;
```

Comment:
as discussed, i think it would be better to condition `passlenuint` to be less than `INT_MAX`.

Reply:
added